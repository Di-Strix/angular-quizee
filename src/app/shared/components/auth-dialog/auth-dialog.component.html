<h1 mat-dialog-title>Missing Auth!</h1>
<mat-dialog-content>To perform this action you have to be authorized.</mat-dialog-content>
<mat-dialog-actions align="end">
  <button
    mat-button
    mat-dialog-close
  >
    Cancel
  </button>
  <button
    mat-button
    (click)="authService.logIn()"
    color="primary"
  >
    Log In
  </button>
</mat-dialog-actions>
