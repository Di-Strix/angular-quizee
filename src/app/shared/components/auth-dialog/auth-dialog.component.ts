import { Component } from '@angular/core';

import { AuthService } from '../../services/auth.service';

@Component({
  selector: 'app-auth-dialog',
  templateUrl: './auth-dialog.component.html',
  styleUrls: ['./auth-dialog.component.scss'],
})
export class AuthDialogComponent {
  constructor(public authService: AuthService) {}
}
