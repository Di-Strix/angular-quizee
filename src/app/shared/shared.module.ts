import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatLegacyButtonModule as MatButtonModule } from '@angular/material/legacy-button';
import { MatLegacyDialogModule as MatDialogModule } from '@angular/material/legacy-dialog';

import { AuthDialogComponent } from './components/auth-dialog/auth-dialog.component';
import { ClassWhenLoadedDirective } from './directives/class-when-loaded.directive';
import { ContainerRefDirective } from './directives/container-ref.directive';

@NgModule({
  declarations: [AuthDialogComponent, ClassWhenLoadedDirective, ContainerRefDirective],
  imports: [CommonModule, MatDialogModule, MatButtonModule],
  exports: [AuthDialogComponent, ClassWhenLoadedDirective, ContainerRefDirective],
})
export class SharedModule {}
