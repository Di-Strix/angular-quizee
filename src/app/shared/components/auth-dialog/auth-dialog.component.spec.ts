import { HarnessLoader } from '@angular/cdk/testing';
import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
import { Component } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatButtonHarness } from '@angular/material/button/testing';

import { AuthService } from '../../services/auth.service';
import { SharedModule } from '../../shared.module';

import { AuthDialogComponent } from './auth-dialog.component';

jest.mock('../../services/auth.service');

describe('AuthDialogComponent', () => {
  let component: AuthDialogComponent;
  let fixture: ComponentFixture<AuthDialogComponent>;
  let loader: HarnessLoader;
  let authService: AuthService;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [AuthDialogComponent],
      imports: [SharedModule],
      providers: [AuthService],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AuthDialogComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();

    loader = TestbedHarnessEnvironment.loader(fixture);
    authService = TestBed.inject(AuthService) as any;
  });

  it('should have correct layout', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should call login function in auth service on login click', async () => {
    const LoginButton = await loader.getHarness(MatButtonHarness.with({ selector: 'button[color="primary"]' }));

    await LoginButton.click();

    expect(authService.logIn).toBeCalledTimes(1);
  });
});
