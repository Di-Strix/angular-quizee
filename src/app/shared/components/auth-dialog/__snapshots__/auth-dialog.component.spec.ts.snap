// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthDialogComponent should have correct layout 1`] = `
<div
  id="root0"
>
  <h1
    class="mat-mdc-dialog-title mdc-dialog__title"
    mat-dialog-title=""
  >
    Missing Auth!
  </h1>
  <mat-dialog-content
    class="mat-mdc-dialog-content mdc-dialog__content"
  >
    To perform this action you have to be authorized.
  </mat-dialog-content>
  <mat-dialog-actions
    align="end"
    class="mat-mdc-dialog-actions mdc-dialog__actions mat-mdc-dialog-actions-align-end"
  >
    <button
      class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
      mat-button=""
      mat-dialog-close=""
      type="button"
    >
      <span
        class="mat-mdc-button-persistent-ripple mdc-button__ripple"
      />
      <span
        class="mdc-button__label"
      >
         Cancel 
      </span>
      <span
        class="mat-mdc-focus-indicator"
      />
      <span
        class="mat-ripple mat-mdc-button-ripple"
        matripple=""
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </button>
    <button
      class="mdc-button mat-mdc-button mat-primary mat-mdc-button-base"
      color="primary"
      mat-button=""
    >
      <span
        class="mat-mdc-button-persistent-ripple mdc-button__ripple"
      />
      <span
        class="mdc-button__label"
      >
         Log In 
      </span>
      <span
        class="mat-mdc-focus-indicator"
      />
      <span
        class="mat-ripple mat-mdc-button-ripple"
        matripple=""
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </button>
  </mat-dialog-actions>
</div>
`;
